addons/hr_expense/hr_expense.py

```diff
@@ -65,257 +32,219 @@ def _get_currency(self, cr, uid, context=None):
-        if journal_ids:
-            res['value']['journal_id'] = journal_ids[0]
-        return res
-
-    def onchange_employee_id(self, cr, uid, ids, employee_id, context=None):
-        emp_obj = self.pool.get('hr.employee')
-        department_id = False
-        employee_payable_account_id = False
-        company_id = False
-        if employee_id:
-            employee = emp_obj.browse(cr, uid, employee_id, context=context)
-            department_id = employee.department_id.id
-            company_id = employee.company_id.id
-            if employee.address_home_id and employee.address_home_id.property_account_payable:
-                employee_payable_account_id = employee.address_home_id.property_account_payable.id
-        return {'value': {'department_id': department_id, 'company_id': company_id, 'employee_payable_account_id': employee_payable_account_id}}
-
-    def expense_confirm(self, cr, uid, ids, context=None):
-        for expense in self.browse(cr, uid, ids):
-            if not expense.line_ids:
+        return super(hr_expense_sheet, self).unlink()
+
+    @api.onchange('currency_id')
+    def onchange_currency_id(self):
+        account_journal = self.env['account.journal'].search([('type', '=', 'purchase'), ('currency', '=', self.currency_id.id), ('company_id', '=', self.company_id.id)], limit=1)
+        self.journal_id = account_journal.id
+
+    @api.multi
+    def sheet_draft(self):
+        for sheet in self:
+            sheet.line_ids.expense_line_draft()
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def sheet_confirm(self):
+        for sheet in self:
+            if not sheet.line_ids:
                 raise UserError(_('You cannot submit expense which has no expense line.'))
-            if expense.employee_id and expense.employee_id.parent_id.user_id:
-                self.message_subscribe_users(cr, uid, [expense.id], user_ids=[expense.employee_id.parent_id.user_id.id])
-        return self.write(cr, uid, ids, {'state': 'confirm', 'date_confirm': time.strftime('%Y-%m-%d')}, context=context)
-
-    def expense_accept(self, cr, uid, ids, context=None):
-        return self.write(cr, uid, ids, {'state': 'accepted', 'date_valid': time.strftime('%Y-%m-%d'), 'user_valid': uid}, context=context)
-
-    def expense_canceled(self, cr, uid, ids, context=None):
-        return self.write(cr, uid, ids, {'state': 'cancelled'}, context=context)
-
-    def account_move_get(self, cr, uid, expense_id, context=None):
+            if sheet.employee_id and sheet.employee_id.parent_id.user_id:
+                self.message_subscribe_users(user_ids = sheet.employee_id.parent_id.user_id.ids)
+            if False not in [True if expense.state in ['draft'] else False for expense in sheet.line_ids]:
+                sheet.line_ids.write({'state': 'confirm'})
```

Comment:
@mdi-odoo 
If expense sheet is in 'draft' state by "set to draft" button and 'hr manager' is want 'Waiting Approval' state on expense sheet so inside sheet expense's is also 'submitted' state.